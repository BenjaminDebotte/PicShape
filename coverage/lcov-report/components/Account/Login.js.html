<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Account/Login.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/Account</a> Login.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/8</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux'
import { login } from '../../actions/auth';
&nbsp;
export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = { email: '', password: '' };
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleChange(event) {
<span class="cstat-no" title="statement not covered" >      this.setState({ [event.target.name]: event.target.value });</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleLogin(event) {
<span class="cstat-no" title="statement not covered" >    event.preventDefault();</span>
<span class="cstat-no" title="statement not covered" >    this.props.dispatch(login(this.state.email, this.state.password));</span>
  }
&nbsp;
  render() {
    return (
      &lt;div className="ui container"&gt;
        &lt;div className="ui raised segment"&gt;
&nbsp;
            &lt;form className="ui form" onSubmit={this.handleLogin.bind(this)}&gt;
              &lt;legend&gt;Log In&lt;/legend&gt;
              &lt;div className="field"&gt;
                &lt;label htmlFor="email"&gt;Email&lt;/label&gt;
                &lt;input type="email" name="email" id="email" placeholder="Email" autoFocus className="form-control" value={this.state.email} onChange={this.handleChange.bind(this)}/&gt;
              &lt;/div&gt;
              &lt;div className="field"&gt;
                &lt;label htmlFor="password"&gt;Password&lt;/label&gt;
                &lt;input type="password" name="password" id="password" placeholder="Password" className="form-control" value={this.state.password} onChange={this.handleChange.bind(this)}/&gt;
              &lt;/div&gt;
              &lt;div className="field"&gt;&lt;Link to="/forgot"&gt;&lt;strong&gt;Forgot your password?&lt;/strong&gt;&lt;/Link&gt;&lt;/div&gt;
              &lt;button type="submit" className="ui button"&gt;Log in&lt;/button&gt;
            &lt;/form&gt;
          &lt;/div&gt;
        &lt;p className="text-center"&gt;
          Don't have an account? &lt;Link to="/signup"&gt;&lt;strong&gt;Sign up&lt;/strong&gt;&lt;/Link&gt;
        &lt;/p&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return {</span>
    messages: state.messages
  };
};
&nbsp;
export default connect(mapStateToProps)(Login);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Thu Mar 09 2017 10:50:14 GMT+0100 (Romance Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
