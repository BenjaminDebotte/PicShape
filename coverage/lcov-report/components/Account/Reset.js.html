<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/Account/Reset.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/Account</a> Reset.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/8</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { connect } from 'react-redux'
import { resetPassword } from '../../actions/auth';
&nbsp;
export class Reset extends React.Component {
  constructor(props) {
    super(props);
    this.state = { password: '', confirm: '' };
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleChange(event) {
<span class="cstat-no" title="statement not covered" >    this.setState({ [event.target.name]: event.target.value });</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleReset(event) {
<span class="cstat-no" title="statement not covered" >    event.preventDefault();</span>
<span class="cstat-no" title="statement not covered" >    this.props.dispatch(resetPassword(this.state.password, this.state.confirm, this.props.params.token));</span>
  }
&nbsp;
  render() {
    return (
        &lt;div className="ui container"&gt;
        &lt;div className="ui raised segment"&gt;
&nbsp;
            &lt;form onSubmit={this.handleReset.bind(this)} className="ui form"&gt;
              &lt;legend&gt;Reset Password&lt;/legend&gt;
              &lt;div className="field"&gt;
                &lt;label htmlFor="password"&gt;New Password&lt;/label&gt;
                &lt;input type="password" name="password" id="password" placeholder="New password" className="form-control" autoFocus value={this.state.password} onChange={this.handleChange.bind(this)}/&gt;
              &lt;/div&gt;
              &lt;div className="field"&gt;
                &lt;label htmlFor="confirm"&gt;Confirm Password&lt;/label&gt;
                &lt;input type="password" name="confirm" id="confirm" placeholder="Confirm password" className="form-control" value={this.state.confirm} onChange={this.handleChange.bind(this)}/&gt;
              &lt;/div&gt;
              &lt;div className="field"&gt;
                &lt;button type="submit" className="btn btn-success"&gt;Change Password&lt;/button&gt;
              &lt;/div&gt;
            &lt;/form&gt;
          &lt;/div&gt;
        &lt;/div&gt;
    );
  }
}
&nbsp;
const mapStateToProps = <span class="fstat-no" title="function not covered" >(s</span>tate) =&gt; {
<span class="cstat-no" title="statement not covered" >  return state;</span>
};
&nbsp;
export default connect(mapStateToProps)(Reset);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Thu Mar 09 2017 10:50:14 GMT+0100 (Romance Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
