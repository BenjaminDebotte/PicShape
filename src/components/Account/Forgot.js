import React from 'react';
import { connect } from 'react-redux'
import { forgotPassword } from '../../actions/auth';

export class Forgot extends React.Component {
  constructor(props) {
    super(props);
    this.state = { email: '' };
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleForgot(event) {
    event.preventDefault();
    this.props.dispatch(forgotPassword(this.state.email));
  }

  render() {
    return (
      <div className="ui container">
                <div className="ui raised segment">
                    <form className="ui form" onSubmit={this.handleForgot.bind(this)}>
                      <div className="field">
                        <p>Enter your email address below and well send you password reset instructions.</p>
                        <label htmlFor="email">Email</label>
                        <input type="email" name="email" id="email" placeholder="Email" className="form-control" autoFocus value={this.state.email} onChange={this.handleChange.bind(this)}/>
                      </div>
                      <button type="submit" className="ui button">Reset Password</button>
                    </form>
            </div>
        </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    messages: state.messages
  };
};

export default connect(mapStateToProps)(Forgot);
