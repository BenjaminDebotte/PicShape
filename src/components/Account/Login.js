import React from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux'
import { login } from '../../actions/auth';

export class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = { email: '', password: '' };
  }

  handleChange(event) {
      this.setState({ [event.target.name]: event.target.value });
  }

  handleLogin(event) {
    event.preventDefault();
    this.props.dispatch(login(this.state.email, this.state.password));
  }

  render() {
    return (
      <div className="ui container">
        <div className="ui raised segment">

            <form className="ui form" onSubmit={this.handleLogin.bind(this)}>
              <legend>Log In</legend>
              <div className="field">
                <label htmlFor="email">Email</label>
                <input type="email" name="email" id="email" placeholder="Email" autoFocus className="form-control" value={this.state.email} onChange={this.handleChange.bind(this)}/>
              </div>
              <div className="field">
                <label htmlFor="password">Password</label>
                <input type="password" name="password" id="password" placeholder="Password" className="form-control" value={this.state.password} onChange={this.handleChange.bind(this)}/>
              </div>
              <div className="field"><Link to="/forgot"><strong>Forgot your password?</strong></Link></div>
              <button type="submit" className="ui button">Log in</button>
            </form>
          </div>
        <p className="text-center">
          Don't have an account? <Link to="/signup"><strong>Sign up</strong></Link>
        </p>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    messages: state.messages
  };
};

export default connect(mapStateToProps)(Login);
