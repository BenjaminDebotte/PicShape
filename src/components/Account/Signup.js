import React from 'react';
import { Link } from 'react-router';
import { connect } from 'react-redux';
import { signup } from '../../actions/auth';

class Signup extends React.Component {
  constructor(props) {
    super(props);
    this.state = { name: '', email: '', password: '' };
  }

  handleChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSignup(event) {
    event.preventDefault();
    this.props.dispatch(signup(this.state.name, this.state.email, this.state.password));
  }


  render() {
    return (
        <div className="ui container">
        <div className="ui raised segment">

            <form onSubmit={this.handleSignup.bind(this)} className="ui form">
              <legend>Create an account</legend>
              <div className="field">
                <label htmlFor="name">Nickname</label>
                <input type="text" name="name" id="name" placeholder="Name" autoFocus className="form-control" value={this.state.name} onChange={this.handleChange.bind(this)}/>
              </div>
              <div className="field">
                <label htmlFor="email">Email</label>
                <input type="email" name="email" id="email" placeholder="Email" className="form-control" value={this.state.email} onChange={this.handleChange.bind(this)}/>
              </div>
              <div className="field">
                <label htmlFor="password">Password</label>
                <input type="password" name="password" id="password" placeholder="Password" className="form-control" value={this.state.password} onChange={this.handleChange.bind(this)}/>
              </div>
              <div className="field">
                <small className="text-muted">By signing up, you agree to the <Link to="/">Terms of Service</Link>.</small>
              </div>
              <button type="submit" className="ui button">Create an account</button>
            </form>
          </div>
        <p className="text-center">
          Already have an account? <Link to="/login"><strong>Log in</strong></Link>
        </p>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    messages: state.messages
  };
};

export default connect(mapStateToProps)(Signup);
