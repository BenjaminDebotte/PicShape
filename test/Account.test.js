import React from 'react';

import {Forgot} from '../src/components/Account/Forgot';
import {Login} from '../src/components/Account/Login';
import {Profile} from '../src/components/Account/Profile';
import {Reset} from '../src/components/Account/Reset';
import {Signup} from '../src/components/Account/Signup';


import renderer from 'react-test-renderer';

describe('components', function() {
  describe('<Forgot />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Forgot />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });

  describe('<Login />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Login />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });

  describe('<Profile />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Profile />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });

  describe('<Reset />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Reset />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });

  describe('<Signup />', function() {
    it('renders correctly', function() {
      var tree = renderer.create(<Signup />).toJSON();
      expect(tree).toMatchSnapshot();
        });
  });
});
