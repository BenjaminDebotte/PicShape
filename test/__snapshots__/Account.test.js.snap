exports[`components <Forgot /> renders correctly 1`] = `
<div
  className="ui container">
  <div
    className="ui raised segment">
    <form
      className="ui form"
      onSubmit={[Function bound handleForgot]}>
      <div
        className="field">
        <p>
          Enter your email address below and well send you password reset instructions.
        </p>
        <label
          htmlFor="email">
          Email
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="email"
          name="email"
          onChange={[Function bound handleChange]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <button
        className="ui button"
        type="submit">
        Reset Password
      </button>
    </form>
  </div>
</div>
`;

exports[`components <Login /> renders correctly 1`] = `
<div
  className="ui container">
  <div
    className="ui raised segment">
    <form
      className="ui form"
      onSubmit={[Function bound handleLogin]}>
      <legend>
        Log In
      </legend>
      <div
        className="field">
        <label
          htmlFor="email">
          Email
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="email"
          name="email"
          onChange={[Function bound handleChange]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div
        className="field">
        <label
          htmlFor="password">
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function bound handleChange]}
          placeholder="Password"
          type="password"
          value="" />
      </div>
      <div
        className="field">
        <a
          onClick={[Function bound handleClick]}
          style={Object {}}>
          <strong>
            Forgot your password?
          </strong>
        </a>
      </div>
      <button
        className="ui button"
        type="submit">
        Log in
      </button>
    </form>
  </div>
  <p
    className="text-center">
    Don\'t have an account? 
    <a
      onClick={[Function bound handleClick]}
      style={Object {}}>
      <strong>
        Sign up
      </strong>
    </a>
  </p>
</div>
`;

exports[`components <Reset /> renders correctly 1`] = `
<div
  className="ui container">
  <div
    className="ui raised segment">
    <form
      className="ui form"
      onSubmit={[Function bound handleReset]}>
      <legend>
        Reset Password
      </legend>
      <div
        className="field">
        <label
          htmlFor="password">
          New Password
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="password"
          name="password"
          onChange={[Function bound handleChange]}
          placeholder="New password"
          type="password"
          value="" />
      </div>
      <div
        className="field">
        <label
          htmlFor="confirm">
          Confirm Password
        </label>
        <input
          className="form-control"
          id="confirm"
          name="confirm"
          onChange={[Function bound handleChange]}
          placeholder="Confirm password"
          type="password"
          value="" />
      </div>
      <div
        className="field">
        <button
          className="btn btn-success"
          type="submit">
          Change Password
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`components <Signup /> renders correctly 1`] = `
<div
  className="ui container">
  <div
    className="ui raised segment">
    <form
      className="ui form"
      onSubmit={[Function bound handleSignup]}>
      <legend>
        Create an account
      </legend>
      <div
        className="field">
        <label
          htmlFor="name">
          Nickname
        </label>
        <input
          autoFocus={true}
          className="form-control"
          id="name"
          name="name"
          onChange={[Function bound handleChange]}
          placeholder="Name"
          type="text"
          value="" />
      </div>
      <div
        className="field">
        <label
          htmlFor="email">
          Email
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          onChange={[Function bound handleChange]}
          placeholder="Email"
          type="email"
          value="" />
      </div>
      <div
        className="field">
        <label
          htmlFor="password">
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[Function bound handleChange]}
          placeholder="Password"
          type="password"
          value="" />
      </div>
      <div
        className="field">
        <small
          className="text-muted">
          By signing up, you agree to the 
          <a
            onClick={[Function bound handleClick]}
            style={Object {}}>
            Terms of Service
          </a>
          .
        </small>
      </div>
      <button
        className="ui button"
        type="submit">
        Create an account
      </button>
    </form>
  </div>
  <p
    className="text-center">
    Already have an account? 
    <a
      onClick={[Function bound handleClick]}
      style={Object {}}>
      <strong>
        Log in
      </strong>
    </a>
  </p>
</div>
`;
